### ISSUE TITLE
Pay Transaction fees using different outputs?

### ISSUE THREAD

**Issue by mtbitcoin** [CounterpartyXCP/counterpartyd#649]
At the moment the bitcoin transaction fees have to come from the "source" address unspent outputs. 

Are there plans to allow payment of bitcoin transaction fees using a non source address.? This would reduce the number of transactions required to consolidate balances from a large list of addresses.

**Comment by adamkrellenstein** [comment]
Plans? Yes. The question is how we'll allow transactions with multiple sources to be parsed. This is a low-level protocol feature, and a security issue, so it'd be good for us to have a powerful reason to make any changes here.

**Comment by mtbitcoin** [comment]
One way I see this is to create an option for specifying a "-feesource" address and use just that address outputs for the fee, its probably more straightforward this way. 

Right now if counterparty is used for token payments/tickets/coupouns, etc, there is a need to topup the new address with some dust BTC first to cover transaction fees when consolidating the token. This option however reduces the number of transactions and time required. 

The recently launched getgems also faced a similar issue with the tokens because new users that had received the GEMZ tokens were unable to spend it because they had no BTC on their addresses . An option for specifying different source address could allow an operator/wallet to bear the transaction fees while still allowing end users full control of their private keys. There are other use cases, but these are the obvious ones that I see

**Comment by SouKw** [comment]
What is the security issue? Why does the BTC fee have to come from the same address as the address sending XCP?

**Comment by mtbitcoin** [comment]
@SouKw  I don't think it's so much a security issue but the complexity and additional testing required to make sure this works correctly

**Comment by adamkrellenstein** [comment]
Yep

**Comment by karlkeefer** [comment]
This would be pretty awesome - we are using tokens in HD wallets so managing the dust for sends adds a lot of complexity to certain operations

**Comment by ninjaboon** [comment]
would love to see this feature implemented.

**Comment by gemamegantara** [comment]
i agree for the implementation of this feature, we have the same problem with mtbitcoin.
our new users who got promotional token/asset in our platform cant spend their asset, because their newly created bitcoin address does not have any bitcoin for it to pay for the tx fee. 

i was wondering whether anyone have a work around for this? appreciate it.

**Comment by mtbitcoin** [comment]
The workaround is to pre-fund the addresses with some dust when you send the tokens over... 

Which bring us to the next feature request of being able to also send btc over during an an asset send. 

At the moment, what you can probably do is to break it up into two transactions. One an asset send and the other a btc send with some "dust btc" to cover the initial transactions. This is your workaround for now because none of these features will make it before your "coinbase hackaton" deadline :-)

**Comment by adamkrellenstein** [comment]
You can already send over as much BTC as you want along with the asset send. Set `--regular-dust-size=$amount`.

**Comment by mtbitcoin** [comment]
Thanks Adam! Great to know that is now possible, but is this also available via the APIs. (i.e create_send) or is this only available via the cli ?

http://counterparty.io/docs/api/#create_send

**Comment by deweller** [comment]
Yes.  This is available via the API.  I use it all the time to use custom dust sizes.

Use regular_dust_size when passing it in via the API (just as it is listed at http://counterparty.io/docs/api/#create_send)

**Comment by mtbitcoin** [comment]
Ahh.. But I am not seeing this in the docs (http://counterparty.io/docs/api/#create_send) though

**Comment by deweller** [comment]
You're right - `regular_dust_size` is missing from the official docs.

Feel like making a pull request?  :smile:

**Comment by mtbitcoin** [comment]
Sure, will try.. whats the default "regular_dust_size" ?

**Comment by deweller** [comment]
Here are the 3 relevant settings:
```
fee_per_kb
regular_dust_size
multisig_dust_size
```

And here are the defaults (in satoshis)
```
DEFAULT_FEE_PER_KB = 10000
DEFAULT_REGULAR_DUST_SIZE = 5430
DEFAULT_MULTISIG_DUST_SIZE = 7800
```

Note that `multisig_dust_size` doesn't apply to OP_RETURN transactions which are the default type used for sends now.

**Comment by unsystemizer** [comment]
@deweller do recent changes change anything related to this issue?

**Comment by longhoangwkm** [comment]
@deweller is this feature finished ?!, l got this problems too
`Pay Transaction Fees with Different Inputs (Multiple Source Addresses) `
